### PR TITLE
[Empty State] Add full width prop

### PR DESCRIPTION
**Pull request by ashwinnarayanan2001**
### WHY are these changes introduced?

Fixes: https://github.com/Shopify/deliveries/issues/206

### WHAT is this pull request doing?
This change is adding another empty state component with a full width property. This full width property works with the centered layout property on the empty state component to provide the user with full width of the content. 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

I've added an example in the README.txt file that shows how the property is supposed to be used.

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated the component's `README.md` with documentation changes
* [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [x] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Comment by ashwinnarayanan2001** [comment]
Looking for feedback on whether I should add padding to the content or if we should leave it to the user. 

Screen Shot 2020-05-20 at 11 52 42 AM

File: src/components/EmptyState/EmptyState.scss [hunk 1]
```diff
@@ -142,6 +142,7 @@ $centered-illustration-width: rem(226px);
 
 .centeredLayout {
   .Section {
+    left: 0;
```
Comment (alex-page) [inline src/components/EmptyState/EmptyState.scss:145]: This looks out of place 🤔

File: src/components/EmptyState/README.md [hunk 2]
```diff
@@ -211,7 +211,7 @@ Use to provide additional but non-critical context for a new product or feature.
     >
       <p>
         You can use the Files section to upload images, videos, and other
-        documents
+        documents!!! This is where we're doing the changes
```
Comment (alex-page) [inline src/components/EmptyState/README.md:214]: Lets revert this example and add more text in the below one!

File: src/components/EmptyState/EmptyState.scss (hunk 1 above)
Comment (kyledurand) [reply src/components/EmptyState/EmptyState.scss:145]: I missed this override in my centeredLayout changes. Not overriding the left property was causing the centeredLayout to be pushed to the right by 2rems or something like that. Ashwin is fixing it for me in this PR

File: UNRELEASED.md [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
 ### Enhancements
 
+- Added a full width property to empty state component with centered layout ([#2992](https://github.com/Shopify/polaris-react/pull/2992))
```
Comment (chloerice) [inline UNRELEASED.md:9]:
```suggestion
- Added a `fullWidth` prop to `EmptyState` to support full width content within a content context ([#2992](https://github.com/Shopify/polaris-react/pull/2992))
```

File: src/components/EmptyState/EmptyState.tsx [hunk 4]
```diff
@@ -23,6 +23,8 @@ export interface EmptyStateProps {
    * Whether or not to limit the image to the size of its container on large screens.
    */
   imageContained?: boolean;
+  /** Overrides the max-width on the empty state with centered layout within content container */
```
Comment (chloerice) [inline src/components/EmptyState/EmptyState.tsx:26]:
```suggestion
  /** Whether or not the content should span the full width of its container  */
```

File: src/components/EmptyState/README.md [hunk 5]
```diff
@@ -237,6 +237,29 @@ Stacked image over centered content and actions
 </EmptyState>
 ```
 
+### Empty state within a content context with centered layout and full width
```
Comment (chloerice) [inline src/components/EmptyState/README.md:240]: When in a content context, the layout is always centered. So we don't need to mention or add the `centeredLayout` prop in this example.

```suggestion
### Empty state with full width layout in a content context 
```

File: src/components/EmptyState/README.md [hunk 6]
```diff
@@ -237,6 +237,29 @@ Stacked image over centered content and actions
...
+
+```jsx
+<Card>
+  <Card.Section>
+    <EmptyState
+      heading="Upload a file to get started"
+      action={{content: 'Upload files'}}
+      image="https://cdn.shopify.com/s/files/1/2376/3301/products/emptystate-files.png"
+      fullWidth
+      centeredLayout
```
Comment (chloerice) [inline src/components/EmptyState/README.md:252]:
```suggestion
```

File: src/components/EmptyState/EmptyState.scss [hunk 7]
```diff
@@ -162,6 +163,10 @@ $centered-illustration-width: rem(226px);
     text-align: center;
     width: 100%;
   }
+
+  &.fullWidth .Details {
+    max-width: 100%;
+  }
 }
```
Comment (chloerice) [inline src/components/EmptyState/EmptyState.scss:170]: We need `fullWidth` to be independent from the `centeredLayout` prop.

```suggestion
}

&.fullWidth {
  .Details {
    max-width: 100%;
  }
}
```

**Comment by chloerice** [comment]
> Looking for feedback on whether I should add padding to the content or if we should leave it to the user.
> 
> Screen Shot 2020-05-20 at 11 52 42 AM

We leave it to the user 👍  In this case, since the empty state is in a card, the implementor (and us in the case of the `fullWidth` example) can use a `Card.Section` or add the `sectioned` prop to `Card` to get the desired spacing around the empty state.

**Pull request merged**